tests/template_tests/tests.py

```diff
@@ -100,3 +105,64 @@ def test_extends_generic_template(self):
         child = engines['django'].from_string(
             '{% extends parent %}{% block content %}child{% endblock %}')
         self.assertEqual(child.render({'parent': parent}), 'child')
+
+
+class VariableResolveLoggingTests(SimpleTestCase):
+    def setUp(self):
+        self.test_handler = utils.TestHandler()
+        self.logger = logging.getLogger('django.template')
+        self.original_level = self.logger.level
+        self.logger.addHandler(self.test_handler)
+        self.logger.setLevel(logging.DEBUG)
+
+    def tearDown(self):
+        self.logger.removeHandler(self.test_handler)
+        self.logger.level = self.original_level
+
+    def test_log_on_variable_does_not_exist_silent(self):
+        class TestObject(object):
+            class SilentDoesNotExist(Exception):
+                silent_variable_failure = True
+
+            @property
+            def template_name(self):
+                return "template"
+
+            @property
+            def template(self):
+                return Template('')
+
+            @property
+            def article(self):
+                raise TestObject.SilentDoesNotExist("Attribute does not exist.")
+
+            def __iter__(self):
+                return iter([attr for attr in dir(TestObject) if attr[:2] != "__"])
+
+            def __getitem__(self, item):
+                return self.__dict__[item]
+
+        with self.settings(DEBUG=True):
+            c = TestObject()
+
+            Variable('article').resolve(c)
+            self.assertEqual(
+                self.test_handler.log_record.msg,
+                'template - Attribute does not exist.'
+            )
+
+    def test_log_on_variable_does_not_exist_not_silent(self):
+        with self.settings(DEBUG=True):
```

Comment:
prefer `@override_settings` if the entire test uses it to avoid an extra level of indentation